adobe/spectrum-web-components
# fix(overlay): ensure smooth animation when opening modal overlays

## Description

Fixes an issue on Safari both iOS Mobile, Tablet and Web where the overlay would open right away, without an animation.
This reproduces on all overlays which use `type=modal`: Tray, Modal dialog, Picker.

Browser version used:
- Safari iOS Mobile 17.6.1
- Safari iOS Tablet 17.5.1
- Safari Web 17.5


**Tray before and after**

https://github.com/user-attachments/assets/5c18342c-e9fe-490d-af31-3b7d98c02aee

**Picker before**

https://github.com/user-attachments/assets/ad296df1-06ff-49b4-a24d-d8c2634022c7


**Picker after**


https://github.com/user-attachments/assets/b97d8ffe-1931-4f0e-b002-ebffbdd5c811



## Related issue(s)



- Fixes https://github.com/adobe/spectrum-web-components/issues/4819

## Motivation and context

See https://github.com/adobe/spectrum-web-components/issues/4819 for context

## How has this been tested?




    1. On Safari iOS, Safari Tablet and Safari Web 
    2. Go to the Tray documentation: https://rocss-4819--spectrum-web-components.netlify.app/components/tray/
    3. Open and close the tray
    4. Both open and close animation should be smooth

    1. On Safari iOS, Safari Tablet and Safari Web 
    2. Go to the Picker documentation: https://rocss-4819--spectrum-web-components.netlify.app/components/picker/
    3. Open and close the Picker
    4. Both open and close animation should be smooth

    1. On Safari iOS, Safari Tablet and Safari Web 
    2. Go to the Dialog base documentation: https://rocss-4819--spectrum-web-components.netlify.app/components/dialog-base/
    3. Open and close the Dialog
    4. Both open and close animation should be smooth


## Screenshots (if appropriate)

## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
